### PR TITLE
fix: [workspace]drag optimization

### PR DESCRIPTION
**Pull request by Lighto-Ku**
1.fetch urls from mimedata when drag enter and cache them for dragMove. 
2.add block mount report.

Log: drag optimization
Bug: https://pms.uniontech.com/bug-view-194099.html

### CODE REVIEW

File: src/plugins/common/dfmplugin-burn/utils/discstatemanager.cpp [hunk 1]
```diff
@@ -56,7 +58,8 @@ void DiscStateManager::onDevicePropertyChanged(const QString &id, const QString
         qint64 curAvial { qvariant_cast<qint64>(map[DeviceProperty::kSizeFree]) };
 
         if (blank && curAvial == 0) {
-            DevMngIns->mountBlockDevAsync(id, {}, [id](bool, const DFMMOUNT::OperationErrorInfo &, const QString &) {
+            DevMngIns->mountBlockDevAsync(id, {}, [id](bool ok, const DFMMOUNT::OperationErrorInfo &, const QString &) {
+                dpfSignalDispatcher->publish("dfmplugin_burn", "signal_ReportLog_BlockMountData", QVariantMap({{"dev", id}, {"result", ok}}));
```
Comment (Johnson-zs) [inline src/plugins/common/dfmplugin-burn/utils/discstatemanager.cpp]: 不太建议在每个插件发送挂载埋点的方法，最好统一处理

**Review by Johnson-zs — CHANGES_REQUESTED** [review]

File: src/dfm-base/base/device/devicemanager.h [hunk 2]
```diff
@@ -106,6 +106,7 @@ class DeviceManager : public QObject
     void blockDevUnlocked(const QString &id, const QString &cleartextId);
     void blockDevFsAdded(const QString &id);
     void blockDevFsRemoved(const QString &id);
+    void aboutToBlockDevMount(const QString &id, bool result);
```
Comment (itsXuSt) [inline src/dfm-base/base/device/devicemanager.h]: blockDevAboutToMount

**Review by itsXuSt — CHANGES_REQUESTED** [review]

File: src/dfm-base/base/device/devicemanager.cpp [hunk 3]
```diff
@@ -153,18 +153,26 @@ void DeviceManager::mountBlockDevAsync(const QString &id, const QVariantMap &opt
...
 
     if (dev->optical()) {
         if (d->isMountingOptical) {
             qWarning() << "Currently mounting a disc!";
+            Q_EMIT aboutToBlockDevMount(id, false);
             return;
         }
+
+        auto callback = [cb, id, this](bool ok, const OperationErrorInfo &err, const QString &mpt) {
+            this->aboutToBlockDevMount(id, ok);
```
Comment (Johnson-zs) [inline src/dfm-base/base/device/devicemanager.cpp]: +emit

**Review by Johnson-zs — APPROVED** [review]

**Review by max-lvs — APPROVED** [review]
+2

**Review by itsXuSt — APPROVED** [review]

**Pull request merged**